### PR TITLE
Docs: Add Configure refresh token handling separately for OAuth providers to what's new 10.1

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What is this feature?**

What's new section about #71533 

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/whatsnew/whats-new-in-v10-1.md [hunk 1]
```diff
@@ -39,3 +39,20 @@ You can now choose whether to set a threshold above which values in the data sho
...
+## Authentication: Configure refresh token handling separately for OAuth providers
+
+<!-- Mihaly Gyongyosi -->
+
+_Available in public preview in all editions of Grafana._
+
+With Grafana v9.3 we introduced a feature toggle called `accessTokenExpirationCheck` to improve the security of Grafana by checking the expiration of the access token and automatically refreshing the expired access token when the user is logged in using one of the OAuth providers.
+
+With the current release, we introduce a new configuration option for each OAuth provider called `use_refresh_token` that allows you to configure whether the particular OAuth integration should use refresh tokens to automatically refresh the access tokens when it expires. Besides, to further improve the security and provide secure defaults we enabled `use_refresh_token` by default for providers that either support refresh tokens by default or fetching refresh tokens is controlled by the client. It is enabled by default for the following OAuth providers: `AzureAD`, `GitLab`, `Google`.
```
Comment (imatwawana) [inline docs/sources/whatsnew/whats-new-in-v10-1.md:51]: This part in the second sentence isn't clear: "refresh tokens by default" - here, do you mean the type of token or the action of refreshing?


```suggestion
With the current release, we've introduced a new configuration option for each OAuth provider called `use_refresh_token`, which allows you to configure whether the particular OAuth integration should use refresh tokens to automatically refresh the access tokens when it expires. In addition, to further improve the security and provide secure defaults, we've enabled `use_refresh_token` by default for providers that support either refresh tokens by default or fetching refresh tokens controlled by the client. It is enabled by default for the following OAuth providers: `AzureAD`, `GitLab`, `Google`.
```

File: docs/sources/whatsnew/whats-new-in-v10-1.md [hunk 2]
```diff
@@ -39,3 +39,20 @@ You can now choose whether to set a threshold above which values in the data sho
...
+
+For more information on how to setup refresh token handling, please refer to [the documentation of the particular OAuth provider.]({{< relref "../setup-grafana/configure-security/configure-authentication/" >}}).
```
Comment (imatwawana) [inline docs/sources/whatsnew/whats-new-in-v10-1.md:53]:
```suggestion
For more information on how to set up refresh token handling, please refer to [the documentation of the particular OAuth provider.]({{< relref "../setup-grafana/configure-security/configure-authentication/" >}}).
```

**Review by imatwawana — COMMENTED** [review]
Thanks for the contribution! Mostly I've made minor wording and formatting edits, but there's one section where I'm not clear on what's being said, so I've left a question and a suggested edit.

File: docs/sources/whatsnew/whats-new-in-v10-1.md (hunk 1 above)
Comment (linoman) [reply docs/sources/whatsnew/whats-new-in-v10-1.md:51]: Hello @imatwawana. I believe @mgyongyosi means the action of refreshing the tokens.

Would it be better rephrased as
```
In addition, to further improve the security and provide secure defaults, we've enabled `use_refresh_token` by default for providers that support either refreshing tokens automatically or fetching refresh tokens controlled by the client. It is enabled by default for the following OAuth providers: `AzureAD`, `GitLab`, `Google`.
```

Let me know what you think.

Comment (imatwawana) [reply docs/sources/whatsnew/whats-new-in-v10-1.md:51]: Yes, that works. Also I still want to rephrase that whole sentence a bit. The second part of the sentence wasn't quite right either. I'm thinking:

`
In addition, to further improve the security and provide secure defaults, `use_refresh_token` is enabled by default for providers that support either refreshing tokens automatically or client-controlled fetching of refresh tokens. 
`

Comment (linoman) [reply docs/sources/whatsnew/whats-new-in-v10-1.md]: Added!

File: docs/sources/whatsnew/whats-new-in-v10-1.md [hunk 3]
```diff
@@ -39,3 +39,23 @@ You can now choose whether to set a threshold above which values in the data sho
...
+
+### Configure refresh token handling separately for OAuth providers
+
+<!-- Mihaly Gyongyosi -->
+
+_Available in public preview in all editions of Grafana._
+
+With Grafana v9.3, we introduced a feature toggle called `accessTokenExpirationCheck`. It improves the security of Grafana by checking the expiration of the access token and automatically refreshing the expired access token when the user is logged in using one of the OAuth providers.
+
+With the current release, we introduce a new configuration option for each OAuth provider called `use_refresh_token` that allows you to configure whether the particular OAuth integration should use refresh tokens to automatically refresh the access tokens when it expires. In addition, to further improve the security and provide secure defaults, `use_refresh_token` is enabled by default for providers that support either refreshing tokens automatically or client-controlled fetching of refresh tokens. It is enabled by default for the following OAuth providers: `AzureAD`, `GitLab`, `Google`.
```
Comment (imatwawana) [inline docs/sources/whatsnew/whats-new-in-v10-1.md:53]:
```suggestion
With the current release, we introduce a new configuration option for each OAuth provider called `use_refresh_token` that allows you to configure whether the particular OAuth integration should use refresh tokens to automatically refresh access tokens when they expire. In addition, to further improve security and provide secure defaults, `use_refresh_token` is enabled by default for providers that support either refreshing tokens automatically or client-controlled fetching of refresh tokens. It's enabled by default for the following OAuth providers: `AzureAD`, `GitLab`, `Google`.
```

**Review by imatwawana — APPROVED** [review]
A couple final wording edits, but otherwise LGTM!

**Pull request merged**